tools/reviewer/web/src/app/dashboard/reviews-panel/review/review.component.ts

```diff
@@ -67,26 +62,31 @@ export class ReviewComponent implements OnInit {
 
   // Save needAttentionOf list and update the Diff
   saveAttentionList(name: string): void {
-    if (this.diff.needAttentionOf.includes(name)) {
-      this.diff.needAttentionOf = this.diff.needAttentionOf.filter(
-        e => e !== name
-      );
+    if (this.diffView.needAttentionOfList.includes(name)) {
+      const filtheredNeedAttentionOfList = this.diff
+        .getNeedAttentionOfList()
+        .filter(e => e !== name);
+
+      this.diff.setNeedAttentionOfList(filtheredNeedAttentionOfList);
     } else {
-      this.diff.needAttentionOf.push(name);
+      this.diff.addNeedAttentionOf(name);
     }
     this.updateDiff(this.diff, 'Update Need Attention List');
   }
 
-  // Remove a property from Diff and update the Diff
+  // TODO: refactor this method
+  // Update whole diff, instead of sending a property to firebase
+  // NOTICE: do not use: object[property]
   removeProperty(property: string, message: string): void {
-    this.firebaseService
-      .removeProperty(this.diff, property)
-      .then(() => {
-        this.notificationService.success(message);
-      })
-      .catch(() => {
-        this.notificationService.error('Some error occured');
-      });
+    console.warn('Unsupported behavior');
```

Comment:
What functionality is now broken?

Reply:
When proto binary will be used, we won't able to edit a specific property.
So basically we need to write a new method which edits local diff and sends it to the server. I left TODO comment about it.

Reply:
Sorry If I missed a discussion but we came to the opinion that updating the whole diff is too broad, I think that's why we used to remove specific property?

Reply:
Right. I was talking about the idea in my last diff. It'd be simpler to delete a specific comment, instead of updating whole diff.
But since we're going to use proto binary, that doesn't make a sense. We have to update a whole diff.